python/lsst/pipe/tasks/findSatellites.py

```diff
@@ -0,0 +1,672 @@
+            if (clusterStandardDeviations <= 1).all():
+                break
+            nClusters += 1
+
+        # The cluster centers are final line estimates
+        finalClusters = kmeans.cluster_centers_.T
+
+        # Sum votes from lines going into clusters:
+        clusterVotes = np.zeros(nClusters)
+        for c in range(nClusters):
+            ind = kmeans.labels_ == c
+            clusterVotes[c] = lines['votes'][ind].sum()
+        finalClustersInfo = np.vstack([finalClusters, clusterVotes])
+        result = np.core.records.fromarrays(finalClustersInfo, dtype=[('rho', float), ('theta', float),
+                                                                      ('votes', int)])
+
+        # Rescale cluster centers
+        result['rho'] *= self.config.rhoBin
+        result['theta'] *= self.config.thetaBin
+
+        return result
+
+    def fitProfile(self, lines, maskedImage):
+        """Fit the profile of the satellite trail.
+
+        Given the initial parameters of detected lines, fit a model for the
+        satellite trail to the original (non-binary image). The assumed model
+        is a straight line with a Moffat profile.
+
+        Parameters
+        ----------
+        lines : `np.ndarray`
+            Array of lines with fields "rho" and "theta"
+        maskedImage : `lsst.afw.image.Exposure`
+            Original image to be used to fit profile of satellite trail.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Result struct with components:
+
+            - ``linefits``: np.ndarray with fields "rho", "theta", and "sigma"
+            - ``finalMask``: 2d np.ndarray with detected satellite trails
+                masked out.
+        """
+        invSigmaInit = self.config.invSigma
+        data = maskedImage.image.array
+        weights = maskedImage.variance.array**-1
+        ymax, xmax = data.shape
+        xrange = np.arange(xmax) - xmax / 2.
+        yrange = np.arange(ymax) - ymax / 2.
+        xmesh, ymesh = np.meshgrid(xrange, yrange)
+        mask = (weights != 0) & np.isfinite(weights)
+
+        lineFits = []
+        finalLineMasks = [np.zeros(data.shape, dtype=bool)]
+        for line in lines:
+            lineModel = LineProfile(data, weights, mask, invSigmaInit, line=line)
+            # Skip any lines that do not cover any data (sometimes happens because of chip gaps)
+            if lineModel.lineMaskSize == 0:
+                continue
+
+            fit, chi2, fitSuccess = lineModel.fit(dChi2Tol=self.config.dChi2Tolerance)
+            sigma = 1 / fit.invSigma
+
+            # Initial estimate should be quite close: fit is deemed unsuccessful if rho or theta
+            # change more than the allowed bin in rho or theta:
+            if ((abs(fit.rho - line.rho) > 2 * self.config.rhoBin) or
+               (abs(fit.theta - line.theta) > 2 * self.config.thetaBin)):
+                fitSuccess = False
```

Comment:
`fitSuccess` originally comes from a few lines further up. I wanted to only have on `continue` statement for all fit failures, so that is why I have formatted it this way, if that makes sense.